src/main/java/com/api/controller/RocketController.java

```diff
@@ -26,7 +36,28 @@ public List<RocketDto> getAllRockets(
             @RequestParam(value = "limit", required = false) Integer limitParam,
             @RequestParam(value = "offset", required = false) Integer offsetParam
     ) {
-        return rocketPresenter.getAllRockets(idParam, limitParam, offsetParam);
+
+        List<RocketDto> rocketDtoList = rocketPresenter.getAllRockets(idParam, limitParam, offsetParam);
+        List<SxRocket> sxRocketList = rocketService.getRockets(rocketDtoList);
+        rocketPersistenceService.saveRockets(sxRocketList);
+        return rocketDtoList;
+    }
+
+    @GetMapping("/filter")
+    public List<RocketDto> getRockets(
+            @RequestParam(value = "page", defaultValue = "0") int page,
+            @RequestParam(value = "limit", defaultValue = "10") int limit,
+            @RequestParam(value = "offset", defaultValue = "0") int offset) {
```

Comment:
why do you need both page and offset? they can be derived, skip to the usage of 1 (offset for example)